### PR TITLE
VectorFEMassIntegrator bug fix [matrix-coef-mass-bug-fix]

### PR DESCRIPTION
**Pull request by mlstowell**
A `DenseMatrix` used as a temporary in `VectorFEMassIntegrator::AssembleElementMatrix2` was being incorrectly sized.  This PR also adds to the unit tests so that this bug would have been caught.

This method also supports vector bases derived from H1 and L2 bases.  It may be wise to expand the unit tests further to confirm proper functioning with these bases.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1337](https://github.com/mfem/mfem/pull/1337) | @mlstowell | @tzanio | @delyank + @homijan | 04/18/20 | 05/03/20 | 05/06/20 | |

### CODE REVIEW

**Comment by tzanio** [comment]
Two questions: 1) is this "ready-for-review"; 2) is it critical for mfem-4.1, or can it wait?

**Comment by tzanio** [comment]
ping: @mlstowell

**Comment by mlstowell** [comment]
> ping: @mlstowell

Thanks, @tzanio!  I had forgotten about this one.  I think it's almost ready.

**Comment by delyank** [comment]
This is a problem that we missed, coming from PR #1068. It's a small typo-like mistake and no test could catch it as it seems. It's good that a test is added now. I'll look at it in more detail a bit later but soon.

File: tests/unit/fem/test_3d_bilininteg.cpp [hunk 1]
```diff
@@ -870,6 +896,31 @@ TEST_CASE("3D Bilinear Vector Mass Integrators",
...
+
+                  SparseMatrix * blfT = Transpose(blfw.SpMat());
+                  SparseMatrix * diff = Add(1.0,blf.SpMat(),-1.0,*blfT);
+
+                  REQUIRE( diff->MaxNorm() < tol );
+
+                  delete blfT;
+                  delete diff;
+               }
                SECTION("With Diagonal Matrix Coefficient")
```
Comment (homijan) [inline tests/unit/fem/test_3d_bilininteg.cpp:924]: Would it be possible to add the VectorFE versions for Diagonal Matrix Coefficient tests?

Comment (mlstowell) [reply tests/unit/fem/test_3d_bilininteg.cpp]: Yes, of course.  Thanks for catching this!  I think I found four such cases and added new tests for each.

**Comment by tzanio** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**